### DISCUSSION TITLE
Fireworks show - Design document

### DISCUSSION THREAD

**Discussion by frutiemax** [OpenRCT2/OpenRCT2#17758] [Q&A]
It would be great to have a Fireworks show feature in OpenRCT2. I've done a preliminary design document that shows the general idea and I'd like to know if this is something that would get accepted.
[Fireworks_OpenRCT2_Version1_0.pdf](https://github.com/OpenRCT2/OpenRCT2/files/9277823/Fireworks_OpenRCT2_Version1_0.pdf)

EDIT - It seems the park format is not in json format (see src/openrct2/park/ParkFile.cpp), the document should however give an idea of what additional data would need to be saved in the park file.

**Comment by ocalhoun6** [comment]
Seems like it would be better as a custom ride, maybe? Kind of like a flat ride, with some kind of bleachers/seating/viewing area that peeps queue for and file into, then the fireworks show comes from a 'launcher' element of the ride, and when the show's over, the peeps file their way out. Maybe combine this with a very high chance of any nearby peeps stopping to watch the ride? And make the peep loading/unloading process very slow, so it doesn't go off very often -- so you don't have _constant_fireworks going off. And then all the different options for customizing the fireworks show would be part of the ride's options and operating modes. (Probably the easiest way to implement that would be that instead of fully custom setups, there would be a half dozen or so premade fireworks routines that you could select from by operating mode.) It could even be synced to music by having the ride make custom sounds ... which include the music for that routine. Oh, and of course the operating costs of your fireworks display could just be the running costs of the ride, which might be higher or lower depending on the operating mode/routine selected.

-------

Or maybe this could be achieved by the use of some custom rides/scenery along with a plugin? It just seems like a lot of stuff to be adding to the base game.

------

Anyway, if you do decide to go forward with this, I'd say why not include other kinds of shows within the same framework?

- Stage shows/concerts. Many parks big and small include stage shows or concerts as attractions. [Example.](https://buschgardens.com/williamsburg/-/media/busch-gardens-williamsburg/images/events/summer-celebration/400x250/400x250_bgw_shows_concertseries.ashx?version=1_202207123240&h=250&w=400&la=en&hash=60106CE6D32787422FB7B471DBD24661BFC1D8C7) [Example](https://www.greatadventurehistory.com/GAshows/FrontAdvStage/20050498%20copy.jpg).
- Parades. (Though that might be too difficult to implement, given their more mobile nature.) [Example.](https://www.wfla.com/wp-content/uploads/sites/71/2018/10/parade_1539702723249_59152094_ver1.0.jpg?w=2560&h=1440&crop=1) [Example.](https://www.themeparktourist.com/files/u10837/Parade_Article/America%20on%20Parade.jpg)
- Laser light shows. (Often in combination with fireworks and/or fountain shows.) [Example.](https://sf-static.sixflags.com/wp-content/uploads/laser-show-2-1-1440x720.jpg)
- Water fountain shows. [Example.](https://www.youtube.com/watch?v=DS_Z432yaiE) [Example.](https://www.youtube.com/watch?v=3HQzmNK6qzM) [Example.](https://content.presspage.com/uploads/1877/1920_efteling-birdviewaquanura.jpg?10000)

All of those could use a very similar framework to a fireworks show. A lot of amusement parks use shows like these as major attractions, in addition to their rides ... fireworks shows are just a subset of the kinds of shows that might be used. And in my opinion, being more versatile like this makes the feature more desirable.

**Reply by frutiemax** [reply]
I wouldn't make it as a ride type since I want the user to choose the firework like an object, and there's not really a waiting line for a show. I like the idea of a spectator zone, this way peeps will prefer some locations over others to watch the show. I guess parades and water shows could also be done but I wanted to keep this simple for a first iteration.

**Comment by karst** [comment]
I was thinking it might be cool to have a speaker scenery item that you can link to certain rides and expand the range of the music they're playing (say if you want it for an entire themed area). If this would be implemented, as well as firework shows, you could make it so the shows would override the speakers.
![PXL_20220726_200605480](https://user-images.githubusercontent.com/1516893/183320558-00fa2bd1-c0f9-4f45-9c7f-f41d62690ca0.jpg) 
![odss-8gssg6sm52taa400phd_1](https://user-images.githubusercontent.com/1516893/183320650-597c297f-9d7f-4e4f-b0b1-797c028d4f59.jpg)
![download (5)](https://user-images.githubusercontent.com/1516893/183320651-ae0e5882-27b5-448d-9f9b-1ed54189e2c9.jpeg)

I do think that this would be a best fit if night mode had some better implementation. Maybe someone could port the schematics shaders to latest version and fine-tune it (idk how much is possible with the game engine, but perhaps take inspiration from octopath traveler) https://youtu.be/bJxlCxnmRiU

**Reply by frutiemax** [reply]
I planned to emit a sound when the firework goes off perhaps people could exploit this. In hardware accelerated mode there's is a night mode, the fireworks could act like a light.
